topologies/binding/binding.go

```diff
@@ -299,43 +303,71 @@ func readGnmi(path string) (*gpb.SetRequest, error) {
 	return req, nil
 }
 
-func applyCli(ctx context.Context, bdut *bindpb.Device, b *staticBind) error {
+func applyCLI(ctx context.Context, bdut *bindpb.Device, r resolver) error {
 	vendorConfig := []string{}
 	for _, conf := range bdut.GetConfig().GetCli() {
 		vendorConfig = append(vendorConfig, string(conf))
 	}
 	for _, file := range bdut.GetConfig().GetCliFile() {
-		conf, err := readCli(file)
+		conf, err := readCLI(file)
 		if err != nil {
 			return err
 		}
 		vendorConfig = append(vendorConfig, conf)
 	}
-	dut := b.resv.DUTs[bdut.GetId()]
 	conf := strings.Join(vendorConfig, "\n")
 
-	return dut.PushConfig(ctx, conf, true)
-}
+	if conf == "" {
+		return nil
+	}
 
-func applyGnmi(ctx context.Context, bdut *bindpb.Device, b *staticBind) error {
-	dialer, err := b.r.gnmi(bdut.GetName())
+	dialer, err := r.ssh(bdut.GetName())
 	if err != nil {
 		return err
 	}
-	conn, err := dialer.dialGRPC(ctx)
+	sc, err := dialer.dialSSH()
 	if err != nil {
 		return err
 	}
-	defer conn.Close()
+	defer sc.Close()
```

Comment:
Rather than sc.Close(), do cli.Close() instead.

Currently cli.Close() will just call sc.Close() but the main idea is that the SSH transport is owned by the CLI object, so we should not operate on it anymore once it's given to newCLI().

Reply:
Done!